src/orchestrationclient.ts

```diff
@@ -355,7 +354,11 @@ export class OrchestrationClient {
         return payload;
     }
 
-    private hasValidRequestUrl(request: any): boolean {
-        return request && (request.http && request.http.url || request.url);
+    private hasValidRequestUrl(request: IRequest): boolean {
+        if (request && request.url) {
```

Comment:
why not simplify this to `return request && request.url`?

Reply:
TypeScript complains that the return type is `string` instead of `boolean` in that format. It accepts the slightly more verbose `request !== undefined && request.url !== undefined` at least.